runtime/vm/description.c

```diff
@@ -164,14 +164,19 @@ calculateInstanceDescription( J9VMThread *vmThread, J9Class *ramClass, J9Class *
 	 * set the corresponding description bit for each 
 	 */
 	{
+		/* Check if there are self referencing field offsets being inherited.
+		 * If so, self referencing fields of the derived class should be ignored.
+		 */
+		BOOLEAN inheritedSelfReferencingFields = ramClass->selfReferencingField1 != 0 ? TRUE : FALSE;
+
 		while (walkResult->field) {
 			UDATA slotOffset = walkResult->offset / (referenceSize * slotsPerShapeElement);
 			J9UTF8 *fieldSig = J9ROMFIELDSHAPE_SIGNATURE(walkResult->field);
 			U_8 *fieldSigBytes = J9UTF8_DATA(fieldSig);
 			U_16 fieldSigLength = J9UTF8_LENGTH(fieldSig);
 
 			/* If the field is self referencing then store the offset to it (at most 2). Self referencing fields are to be scanned with priority during GC */
-			if (((ramClass->selfReferencingField1 == 0) || (ramClass->selfReferencingField2 == 0)) && J9UTF8_DATA_EQUALS(J9UTF8_DATA(className), J9UTF8_LENGTH(className), fieldSigBytes + 1, fieldSigLength - 2)) {
+			if (!inheritedSelfReferencingFields && ((ramClass->selfReferencingField1 == 0) || (ramClass->selfReferencingField2 == 0)) && J9UTF8_DATA_EQUALS(J9UTF8_DATA(className), J9UTF8_LENGTH(className), fieldSigBytes + 1, fieldSigLength - 2)) {
```

Comment:
for readability I'd brake this if condition in two nested if conditions. perhpas move the comment from up down to here?